src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -299,36 +309,32 @@ void CARMACloudPlugin::Broadcast_TCMs()
+				PLOG(logINFO) << " CARMACloud Plugin :: Repeatedly Broadcast tsm5:: " << tsm5ENC.get_payload_str();
 			} //END TCMs LOOP
 
 			// For any ids which have expired clean up the maps
 			for (auto tcmv01_req_id_hex : expired_req_ids) {
-
 				//Create an event log object for both NO ACK (ackownledgement), and broadcast the event log
-
-
-					tmx::messages::TmxEventLogMessage event_log_msg;
-					event_log_msg.set_level(IvpLogLevel::IvpLogLevel_warn);
-					event_log_msg.set_description(_TCMAcknowledgementStrategy + ": " + _TCMNOAcknowledgementDescription + " Traffic control id = " + tcmv01_req_id_hex);
-					PLOG(logDEBUG) << "event_log_msg " << event_log_msg << std::endl;
-					this->BroadcastMessage<tmx::messages::TmxEventLogMessage>(event_log_msg);	
-
-					//send negative ack to carma-cloud
-					stringstream sss;
-					sss << "<?xml version=\"1.0\" encoding=\"UTF-8\"?><TrafficControlAcknowledgement><reqid> " << tcmv01_req_id_hex
-						  << "</reqid><msgnum></msgnum><cmvid></cmvid><acknowledgement>" << acknowledgement_status::acknowledgement_status__not_acknowledged
-						  << "</acknowledgement><description>" << _TCMNOAcknowledgementDescription
-						  << "</description></TrafficControlAcknowledgement>"; 
-					PLOG(logINFO) << "Sent No ACK as Time Out: "<< sss.str() <<endl;
-					CloudSend(sss.str(),url, base_ack, method);		
-
-					_not_ACK_TCMs->erase(tcmv01_req_id_hex);
-					//If time out, stop tracking the starting time of the TCMs being broadcast so far
-					_tcm_broadcast_starting_time->erase(tcmv01_req_id_hex);
-					//If time out, stop tracking the number of times the TCMs ( that has the same TCR reqid) being broadcast
-					_tcm_broadcast_times->erase(tcmv01_req_id_hex);			
-
+				tmx::messages::TmxEventLogMessage event_log_msg;
+				event_log_msg.set_level(IvpLogLevel::IvpLogLevel_warn);
+				event_log_msg.set_description(_TCMAcknowledgementStrategy + ": " + _TCMNOAcknowledgementDescription + " Traffic control id = " + tcmv01_req_id_hex);
+				PLOG(logDEBUG) << "event_log_msg " << event_log_msg << std::endl;
+				this->BroadcastMessage<tmx::messages::TmxEventLogMessage>(event_log_msg);	
+				
+				//send negative ack to carma-cloud
+				stringstream sss;
+				sss << "<?xml version=\"1.0\" encoding=\"UTF-8\"?><TrafficControlAcknowledgement><reqid> " << tcmv01_req_id_hex
+						<< "</reqid><msgnum></msgnum><cmvid></cmvid><acknowledgement>" << acknowledgement_status::acknowledgement_status__not_acknowledged
+						<< "</acknowledgement><description>" << _TCMNOAcknowledgementDescription
+						<< "</description></TrafficControlAcknowledgement>"; 
+				PLOG(logINFO) << "Sent No ACK as Time Out: "<< sss.str() <<endl;
+				CloudSend(sss.str(),url, base_ack, method);		
+
+				_not_ACK_TCMs->erase(tcmv01_req_id_hex);
+				//If time out, stop tracking the starting time of the TCMs being broadcast so far
+				_tcm_broadcast_starting_time->erase(tcmv01_req_id_hex);

```

Comment:
Is this deleting all TCMs associated with this TCR or just the particular TCM that this applies to?